File: MetaBrain/providers_py/local/adapter.py
from __future__ import annotations

from MetaBrain.providers_py.provider_health import disabled_provider_health
from MetaBrain.providers_py.provider_response import build_provider_response
from MetaBrain.providers_py.types import ProviderAdapter, ProviderCapabilities, ProviderRequest, ProviderResponse


class DisabledLocalAdapter(ProviderAdapter):
    def __init__(self) -> None:
        super().__init__(
            provider_name="local",
            model_name="local-disabled",
            capabilities=ProviderCapabilities(True, False, False, False, False, False, 2048, True),
        )

    def complete(self, request: ProviderRequest) -> ProviderResponse:
        return build_provider_response(request, provider_name="local", model_name=self.model_name, status="disabled", safety_flags=["LOCAL_DISABLED"])

    def healthcheck(self):
        return disabled_provider_health("local")

File: MB-Whatsapp/providers/openai/adapter.ts
import { buildProviderResponse } from "../provider-response";
import type { ProviderRequest, ProviderResponse } from "../types";
import { OPENAI_CAPABILITIES } from "./capabilities";
import { openAIHealthDisabled } from "./health";
import type { OpenAIProviderAdapter } from "./contract";

export class DisabledOpenAIAdapter implements OpenAIProviderAdapter {
  readonly provider_name = "openai" as const;
  readonly model_name = "openai-disabled";
  readonly capabilities = OPENAI_CAPABILITIES;

  async complete(input: ProviderRequest): Promise<ProviderResponse> {
    return buildProviderResponse({
      request: input,
      provider_name: "openai",
      model_name: this.model_name,
      status: "disabled",
      safety_flags: ["PROVIDER_DISABLED", "NO_EXTERNAL_CALL"],
    });
  }

  async healthcheck() {
    return openAIHealthDisabled();
  }
}
